### ISSUE TITLE
Social charge with user related create unbalanced entry in accounting breakdown

### ISSUE THREAD

**Issue by Amael-PE** [Dolibarr/dolibarr#20527]
### Bug

**Description**
A paid social charge with user related create unbalanced entry in accounting breakdown.
3 accounts are modified instead of 2, and the resulting entry is balanced.
See bellow for details.

**Bypass**
I 
- deleted the bank entry corresponding to the charge
- marked the charge unpayed again
- unlink the user by modifying the charge
- register the payment again
This second time, that is without user related, the accounting breakdown shown a balanced entry, as expected.

### Environment Version

14.0.4

### Environment OS

Linux 5.4.0-88-generic #99-Ubuntu SMP Thu Sep 23 17:29:00 UTC 2021 x86_64

### Environment Web server

Apache

### Environment PHP

 7.4.3

### Environment Database

MySQL or MariaDB 5.5.5-10.3.34-MariaDB-0ubuntu0.20.04.1

### Environment URL(s)

https://mydolibarr.com/htdocs/accountancy/journal/bankjournal.php?mainmenu=accountancy&leftmenu=accountancy_journal&id_journal=3

### Expected and actual behavior

**Actual behaviour**
When a social charge is payed, with a user related, the bank inscription relate to user and charge (as can be expected).
However, when coming to accounting breakdown ("ventilation" in French), things goes wrong: the line is accounted twice, one from the social charge account and one from user account. 
Then, in the Journal the line is : 
> Desc                            | Account                                                                      | Debit | Credit
> BankId - Social Charge | Bank (1010 in CH accounting plan)                             | 0       | 100
> BankId - Social Charge | Social charge debt (2270 in CH accounting plan)         | 100   | 0
> BankId - Social Charge | User personnal account (2000 in CH accounting plan) | 100   | 0
Then, the line in unbalanced and when trying to write it in the "Grand Livre", an error is raised indicated that this is probably a bug. (And it is)

**Expected behaviour**
Even when indicating a user linked to a social tax, the accounting entry should stay balanced, that is no entry should be writen on user account.

### Steps to reproduce the behavior

**Steps to reproduce**
- Create a tax/social charge, and link a user to it
- Mark the social charge as "Paid", by registring the payment
- Go to Accounting > Accounting Breakdown (_Ventilation_ in French) and select the right period to make the charge appears (Bank Journal).
- If bug reproduce, you should have un unbalanced entry. Try to save and rigth it into the accounting ledger (_Grand Livre_) and the error message should appear.

**Comment by aspangaro** [comment]
I confirm this problem.

I reproduce the bug easily :

1/ I have created a social contribution without user and an another with a user declared : 

On the second contribution, a user line is generated on bank journal and the accounting entry is unbalanced and cannot be integrated. The last line cannot be here.

![image](https://user-images.githubusercontent.com/2341395/161695004-68d208cb-aa6d-4a21-a1ba-944c2e62501e.png)


2/ Situation on database (same situation) : 

The last line user cannot be here because the accounting is based on the type "user" to create its counterpart line

![image](https://user-images.githubusercontent.com/2341395/161694671-fe2b1e4b-6bd1-45ac-b999-b218d90eb458.png)

![image](https://user-images.githubusercontent.com/2341395/161695514-322b5697-c0dc-460b-8640-80a519aaedbd.png)

@atm-maxime You have added this feature (link an user to a social contribution), possibility to correct the problem ?

Thanks in advance, I don't see how to solve the problem at the moment

**Comment by daraelmin** [comment]
Similiar to this thread

https://www.dolibarr.fr/forum/t/doublon-dans-le-journal-de-tresorerie/39626

**Comment by Daviid-P** [comment]
There's something I don't understand,

The problem is that if there's a user linked it tries to add another row for user account.

So, I assume this was added to skip the user account link:
`if ($links[$key]['type'] == 'user' && !$is_sc) continue;`

But if It's a social charge `$is_sc` is always true, so we end up with

```
// if ($links[$key]['type'] == 'user' && !$is_sc) continue;
if    (      true                     && !true  ) continue;
if    (      true                     && false  ) continue;
if    (                  false                  ) continue;
```

and never skips?


I ask beacuse I'm having this exact problem, I though I might not have this fix due to being on 16.0.3 but I do have it.

![image](https://github.com/Dolibarr/dolibarr/assets/7734808/0acd282d-e454-48ad-a7b5-8877ef0b6ab1)


Either I'm not understanding something (The point of this commit is to skip the type user on social charges right?), I have made a mistake before the bookkeeping step or this problem was never fixed?
